src/Hooks/useFetchUserData.js

```diff
@@ -0,0 +1,8 @@
+import axios from 'axios'
+import {useQuery} from 'react-query'
+function FetchData(api){
+    return axios.get(api);
+}
+export default function useFetchUserData(Id,api) {
```

Comment:
instead of generic hook, use custom hooks like useGetProducts() and useGetProduct().